cpp/src/arrow/flight/flight_test.cc

```diff
@@ -1556,13 +1556,14 @@ TEST_F(TestBasicAuthHandler, CheckPeerIdentity) {
   ASSERT_EQ(result->body->ToString(), "user");
 }
 
-#ifdef __APPLE__
-// ARROW-7701: this test is flaky on MacOS and segfaults (due to gRPC
-// bug?)
-TEST_F(TestTls, DISABLED_DoAction) {
-#else
 TEST_F(TestTls, DoAction) {
-#endif
+  // Manually initialize gRPC to try to ensure some thread-locals get
+  // initialized.
+  // https://github.com/grpc/grpc/issues/13856
+  // https://github.com/grpc/grpc/issues/20311
+  // In general, gRPC on MacOS struggles with TLS (both in the sense
+  // of thread-locals and encryption)
+  grpc_init();
```

Comment:
Can you move this in the test's fixture setup?

Reply:
Done, thanks for the catch.